athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/storage/StorageMetadata.java

```diff
@@ -0,0 +1,90 @@
+/*-
+ * #%L
+ * athena-hive
+ * %%
+ * Copyright (C) 2019 - 2022 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.gcs.storage;
+
+import com.amazonaws.athena.connector.lambda.domain.TableName;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Constraints;
+import com.amazonaws.athena.connector.lambda.metadata.MetadataRequest;
+import com.amazonaws.athena.connectors.gcs.common.PartitionFolder;
+import com.amazonaws.athena.connectors.gcs.common.PartitionLocation;
+import com.amazonaws.athena.connectors.gcs.storage.datasource.StorageTable;
+import com.amazonaws.services.glue.AWSGlue;
+import com.google.cloud.storage.Storage;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.arrow.dataset.file.FileFormat;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.List;
+import java.util.Optional;
+
+public interface StorageMetadata
```

Comment:
Same comment here. This interface is unnecessary. Delete this file and copy over the comments (wherever its just `inheritdoc`) to the Impl (and then rename the Impl to just StorageMetadata.java).

Reply:
Removed.